src/domain/request/Request.ts

```diff
@@ -16,6 +16,14 @@ class Request {
   getSelectionPath() {
     return RequestUrlParser.resolveSelectionPath(this.url);
   }
+
+  getLeftSideBound() {
```

Comment:
I'm not sure if it's the right place for this. Not every request would have left and right side bounds (but from the code it would look like this).

Reply:
It is not. I will move `RequestUrlParser` logic to RequestStrategy, along with this as well. I intend to run an enhancement task that will address couple of things like this plus add more tests around random provider feature. However I decided to postpone it because we have a feature to deliver.